test/saithrift/tests/sail2.py

```diff
@@ -379,3 +379,94 @@ def runTest(self):
             self.client.sai_thrift_set_port_attribute(port2, attr)
             self.client.sai_thrift_set_port_attribute(port3, attr)
             self.client.sai_thrift_set_port_attribute(port4, attr)
+			
+class L2VlanBcastUcastTest(sai_base_test.ThriftInterfaceDataPlane):
+    def runTest(self):
+        """
+        For SONiC
+        Vlan broadcast and known unicast test. Verify the broacast packet reaches all ports in the vlan and unicast packet reach specific port. 
+        Steps:
+        1. remove all ports from default vlan
+        2. create vlan 10
+        3. add n-1 ports to vlan 10
+        4. add mac for each port
+        5. send untagged broadcast packet from port 1, verify all n-1 ports receive the packet except the last port
+        6. send untagged unicast packets from port 1 to the rest of the vlan members ports. Verify only one port at a time receives the packet and port n does not.
+        7. clean up.
+        """
+
+        switch_init(self.client)
+        default_vlan = 1
+        vlan_id = 10
+        default_vlan_members_list = []
+        mac_list = []
+        vlan_member_list = []
+
+        for i in range (1, len(port_list)):
+            str_i = str(i)
+            mac_list.append('00:'+str_i+str_i+':'+str_i+str_i+':'+str_i+str_i+':'+str_i+str_i+':'+str_i+str_i)
```

Comment:
maybe "00:00:00:00:00:%d" % (i + 1) is better.

then the first port will be 00:00:00:00:00:01 as all zero is invalid.

---
In reply to: [71752680](https://github.com/opencomputeproject/SAI/pull/150#discussion_r71752680) [](ancestors = 71752680)